packages/types/src/index.ts

```diff
@@ -36,146 +36,579 @@ export interface Dialog {
+ * menu, or to supply a list of less visually important actions after a row of
+ * buttons.
+ *
+ * To use interactive elements like this, you will need to make some changes
+ * to prepare your app. Read our
+ * [guide to enabling interactivity](https://api.slack.com/messaging/interactivity/enabling).
+ *
+ * https://api.slack.com/reference/messaging/block-elements#overflow
+ */
 export interface Overflow extends Action {
   type: 'overflow';
+
+  /**
+   * An array of option objects to display in the menu. Maximum number of options
+   * is 5, minimum is 2.
+   */
   options: Option[];
+
+  /**
+   * A plain_text only text object that defines the placeholder text shown on
+   * the menu. Maximum length for the text in this field is 150 characters.
+   */
+  placeholder: PlainTextElement;
 }
 
+/**
+ * An element which lets users easily select a date from a calendar style UI.
+ * Date picker elements can be used inside of section and actions blocks.
+ *
+ * To use interactive elements like this, you will need to make some changes
+ * to prepare your app. Read our guide to enabling interactivity.
+ *
+ * https://api.slack.com/reference/messaging/block-elements#datepicker
+ */
 export interface Datepicker extends Action {
   type: 'datepicker';
+
+  /**
+   * The initial date that is selected when the element is loaded. This
+   * should be in the format `YYYY-MM-DD`.
+   */
   initial_date?: string;
+
+  /**
+   * A plain_text only text object that defines the placeholder text shown
+   * on the menu. Maximum length for the text in this field is 150 characters.
+   */
   placeholder?: PlainTextElement;
 }
 
 /*
  * Block Types
  */
 
+/**
+ * Block elements can be used inside of `section`, `context`, and `actions`
+ * [layout blocks](https://api.slack.com/reference/messaging/blocks).
+ *
+ * Our guide to [composing rich message layouts](https://api.slack.com/messaging/composing/layouts)
+ * shows you where and how to add blocks to messages.
+ *
+ * Finally, the [enabling interactivity guide](https://api.slack.com/messaging/interactivity/enabling)
+ * will help you adjust your app to allow for the use of these interactive components.
+ *
+ * https://api.slack.com/reference/messaging/block-elements
+ */
 export type KnownBlock = ImageBlock | ContextBlock | ActionsBlock | DividerBlock | SectionBlock;
 
+/**
+ * Block elements can be used inside of `section`, `context`, and `actions`
+ * [layout blocks](https://api.slack.com/reference/messaging/blocks).
+ *
+ * Our guide to [composing rich message layouts](https://api.slack.com/messaging/composing/layouts)
+ * shows you where and how to add blocks to messages.
+ *
+ * Finally, the [enabling interactivity guide](https://api.slack.com/messaging/interactivity/enabling)
+ * will help you adjust your app to allow for the use of these interactive components.
+ *
+ * https://api.slack.com/reference/messaging/block-elements
+ */
 export interface Block {
   type: string;
+
+  /*
+   * Undocumented
```

Comment:
Documentation for Block ID is under image block https://api.slack.com/reference/messaging/blocks#image


A optional string acting as a unique identifier for a block. You can use this block_id when you receive an interaction payload to identify the source of the action. If not specified, a block_id will be generated. Maximum length for this field is 255 characters.